### PR TITLE
Bug/jenkins 44432 pipeline editor save

### PR DESCRIPTION
**Pull request by vivek**
# Description

See [JENKINS-44432](https://issues.jenkins-ci.org/browse/JENKINS-44432).

Repost of https://github.com/jenkinsci/blueocean-plugin/pull/1078 with core js released version.

# Submitter checklist
- [ ] Link to JIRA ticket in description, if appropriate.
- [ ] Change is code complete and matches issue description
- [ ] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.
- [ ] Ran Acceptance Test Harness against PR changes.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

### CODE REVIEW

File: blueocean-pipeline-scm-api/src/main/java/io/jenkins/blueocean/scm/api/AbstractPipelineCreateRequest.java [hunk 1]
```diff
@@ -23,12 +26,19 @@
 
     protected final BlueScmConfig scmConfig;
 
-    public AbstractPipelineCreateRequest(String name, String organization, BlueScmConfig scmConfig) {
+    public AbstractPipelineCreateRequest(String name, BlueScmConfig scmConfig) {
         setName(name);
-        setOrganization(organization);
+        Collection<BlueOrganization> organizations = OrganizationFactory.getInstance().list();
```
Comment (cliffmeyers) [inline blueocean-pipeline-scm-api/src/main/java/io/jenkins/blueocean/scm/api/AbstractPipelineCreateRequest.java]: Is this pluggable via an ExtensionPoint?

Comment (michaelneale) [inline blueocean-pipeline-scm-api/src/main/java/io/jenkins/blueocean/scm/api/AbstractPipelineCreateRequest.java]: quite possibly

Comment (vivek) [inline blueocean-pipeline-scm-api/src/main/java/io/jenkins/blueocean/scm/api/AbstractPipelineCreateRequest.java]: @cliffmeyers yes

File: blueocean-core-js/src/js/utils/UrlUtils.js [hunk 2]
```diff
@@ -276,6 +276,12 @@ export function toClassicJobPage(pageUrl, isMultibranch = false) {
         const fullJobNameTokens = fullJobName.split('/');
         const classicJobFullName = '/job/' + fullJobNameTokens.join('/job/');
 
+        // map to job only after there is token after pipelines. For example /organizations/jenkins/pipelines/p1
+        // can be mapped to /job/p1 but /organizations/jenkins/pipelines/ should be mapped to classic dashboard '/'
+        if (fullJobName === 'pipelines' && pageUrlTokens.length === 0) {
+            return undefined;
+        }
```
Comment (cliffmeyers) [inline blueocean-core-js/src/js/utils/UrlUtils.js]: I take it this is really more about fixing an existing bug than anything strictly related to the features here?

Comment (michaelneale) [inline blueocean-core-js/src/js/utils/UrlUtils.js]: yes, there were 2 regressions: https://issues.jenkins-ci.org/browse/JENKINS-44432 && https://issues.jenkins-ci.org/browse/JENKINS-44433 that this is addressing.

It was broken on master but also slightly broken before.

**Comment by michaelneale** [comment]
:bee: seems happier now

**Pull request merged**